onnx_tf/handlers/backend/gather.py

```diff
@@ -12,3 +12,19 @@ class Gather(BackendHandler):
   @classmethod
   def version_1(cls, node, **kwargs):
     return [cls.make_tensor_from_onnx_node(node, **kwargs)]
+
+  @classmethod
+  def version_11(cls, node, **kwargs):
+    x = kwargs["tensor_dict"][node.inputs[0]]
+    indices = kwargs["tensor_dict"][node.inputs[1]]
+    axis = node.attrs.get("axis", 0)
+    x_rank = len(x.shape)
+    if axis < -x_rank or axis >= x_rank:
+      raise ValueError("axis %d is out of range. must be in [%d, %d]" % (axis, -x_rank, x_rank-1))
+    dimension = tf.cast(tf.shape(x)[axis], indices.dtype)
+    indices_shape = tf.shape(indices)
+    dim_broadcasted = tf.broadcast_to(dimension, indices_shape)
+    nonneg_indices = tf.cast(tf.greater_equal(indices, 0), dtype=indices.dtype)
+    negative_indices = tf.cast(tf.less(indices, 0), dtype=indices.dtype)
+    final_indices = dim_broadcasted * negative_indices + negative_indices * indices + nonneg_indices * indices
+    return [tf.gather(x, final_indices, axis=axis, name=cls.tf_node_name(node.name))]
```

Comment:
@winnietsang here are the negative indices support. Note that other op transformation invoking `tf` functions do not set the op name; I set it here, though if that's against the convention I can remove it.

Also please specify what's needed for the error checking. Here I validate the `axis` attribute. For indices, the shape can be undefined and that run-time error will be thrown by tensorflow if the indices are out of range at run-time.

Reply:
@winston-zillow please check indices out of bounds and throw exception before calling tf.gather. 
Tensorflow run in CPU will detect indices out of bounds and throw exception. But in GPU, Tensorflow will not throw exception, it will return a '0' for that corresponding output value. 
In Tensorflow documentation (https://www.tensorflow.org/api_docs/python/tf/gather) it state that this is the expected behavior. While in ONNX documentation (https://github.com/onnx/onnx/blob/master/docs/Operators.md#Gather) whenever there is indices out of bounds need to throw exception.

Reply:
@winnietsang do you see it is sufficient to raise out of bound error when the static shape is known. I wonder if there is a way to get TF to raise an error after checking the dynamic shapes.

Reply:
@winnietsang OK, I saw that you use `tf.compat.v1.assert_equal` in the `gather_nd` op. strange that the `assert_equal` became undocumented since TF 1.12, not even in the `t1.compat`. Will take use of that as well

Reply:
@winston-zillow Yes, I used tf.compat.v1.assert_equal to get Tensorflow to throw exception for me in the gather_nd support, you can reuse all of those code here.